Storage RN 4.16 (#75937)
openshift/openshift-docs: release_notes/ocp-4-16-release-notes.adoc

[id="ocp-4-16-storage"]
=== Storage

[id="ocp-4-16-storage-clone-azure-file"]

> ==== Volume cloning supported for Azure File (Technology Preview)

Reviewer: According to google, there is an "Azure Files" product with an s, but maybe that's something different - just checking.

Author: I believe that that is referring to something else.

> {product-title} 4.16 introduces volume cloning for the Microsoft Azure File Container Storage Interface (CSI) Driver Operator as a Technology Preview feature. Volume cloning duplicates an existing persistent volume to help protect against data loss in {product-title}. 

Reviewer: ```suggestion
{product-title} 4.16 introduces volume cloning for the Microsoft Azure File Container Storage Interface (CSI) Driver Operator as a Technology Preview feature. Volume cloning duplicates an existing persistent volume (PV) to help protect against data loss in {product-title}. 
```

> 

Reviewer 2: I suggest adding a note that the nfs protocol doesn't support volume cloning. (Or make sure we have this somewhere when introducing azure-file volume cloning.)

Reviewer 3: Yeah agree, we need to mention it does not supports nfs protocol, while it supports with default sc which has smb protocol.

Author: I've added this info to the cloning feature content: https://github.com/openshift/openshift-docs/pull/73733

For more information, see xref:..//storage/container_storage_interface/persistent-storage-csi-azure-file.adoc#persistent-storage-csi-azure-file[Azure File CSI Driver Operator] and xref:..//storage/container_storage_interface/persistent-storage-csi-cloning.adoc#persistent-storage-csi-cloning[CSI volume cloning].

[id="ocp-4-16-storage-node-expansion-secret"]
==== Node Expansion Secret is generally available

> The Node Expansion Secret feature allows your cluster to expand storage volumes, even when access to those volumes requires a secret (for example, a credential for accessing a SAN fabric) to perform the node expand operation. {product-title} 4.16 supports this feature as generally available.

Reviewer: should SAN be spelled out at first use?

[id="ocp-4-16-vsphere-max-snapshot"]

> ==== Changing vSphere CSI maximum number of snapshots is generally available

Reviewer 2: If I remember correctly, it is a TP feature, right?
In that case, we should update the TP table accordingly.

Reviewer 2: Ignore me due to: 
https://redhat-internal.slack.com/archives/GK0DA0JR5/p1715754771284149

Reviewer 4: @lpettyjo As Wei pointed out - GA is not yet certain for this feature. I'll keep you posted.

Author: The link to the content is there, but commented out to avoid a publishing error.

> The default maximum number of snapshots per volume in VMWare vSphere Container Storage Interface (CSI) is 3. In {product-title} 4.16, you can now change this maximum number of snapshots to a maximum of 32 per volume. You also have granular control of the maximum number of snapshots for vSAN and Virtual Volume datastores. {product-title} 4.16 supports this feature as generally available.

Reviewer 5: ```suggestion
The default maximum number of snapshots per volume in VMWare vSphere Container Storage Interface (CSI) is 3 per volume. In {product-title} 4.16, you can now change this maximum number of snapshots to a maximum of 32 per volume. You also have granular control of the maximum number of snapshots for vSAN and Virtual Volume datastores. {product-title} 4.16 supports this feature as generally available.
```
Maybe better.

For more information, see xref:..//storage/container_storage_interface/persistent-storage-csi-vsphere.adoc#vsphere-change-max-snapshot_persistent-storage-csi-vsphere[Changing the maximum number of snapshots for vSphere]

[id="ocp-4-16-pv-last-phase-transition-time"]

> ==== Persistent volume last phase transition time parameter (Technology Preview)

Reviewer 6: LGTM

> In {product-title} 4.16 introduces a new parameter `LastPhaseTransitionTime`, which has a timestamp that is updated every time a persistent volume (PV) transitions to a different phase (`pv.Status.Phase`). This feature is being released with Technology Preview status.

Reviewer: ```suggestion
In {product-title} 4.16 introduces a new parameter, `LastPhaseTransitionTime`, which has a timestamp that is updated every time a persistent volume (PV) transitions to a different phase (`pv.Status.Phase`). This feature is being released with Technology Preview status.
```

[id="ocp-4-16-storage-csi-smb-cifs-driver-operator"]
==== Persistent storage using SMB CIFS CSI Driver Operator (Technology Preview)

> {Product-title} is capable of provisioning persistent volumes (PVs) using the Container Storage Interface (CSI) driver for the Server Message Block (SMB) protocol Common Internet File System (CIFS) dialect. The SMB CIFS CSI Driver Operator that manages this driver is in Technology Preview.

Reviewer: ```suggestion
{Product-title} is capable of provisioning PVs using the Container Storage Interface (CSI) driver for the Server Message Block (SMB) protocol Common Internet File System (CIFS) dialect. The SMB CIFS CSI Driver Operator that manages this driver is in Technology Preview status.
```

Reviewer: ^
1) To be consistent with previous way of referring to TP
2) I suggested adding (PV) at first use of persistent volume earlier in doc if you wanted to use just PV here

Reviewer 7: @lpettyjo , I'm OK about this SMB section (lgtm).

// For more information, see xref:..//storage/container_storage_interface/persistent-storage-csi-smb-cifs.adoc#persistent-storage-csi-smb-cifs[SMB CIFS CSI Driver Operator].

[id="ocp-4-16-storage-rwop-selinux-context-mode"]
==== RWOP with SELinux context mount is generally available

> {product-title} 4.14 introduced a new access mode with Technical Preview status for persistent volumes (PVs) and persistent volume claims (PVCs) called ReadWriteOncePod (RWOP). RWOP can be used only in a single pod on a single node compared to the existing ReadWriteOnce access mode where a PV or PVC can be used on a single node by many pods. RWOP uses SELinux context mount, which allows the driver to mount the volume directly with the correct SELinux labels, eliminating the need to recursively relabel the volume, and pod startup can be significantly faster. 

Reviewer: ```suggestion
{product-title} 4.14 introduced a new access mode with Technical Preview status for PVs and persistent volume claims (PVCs) called ReadWriteOncePod (RWOP). RWOP can be used only in a single pod on a single node compared to the existing `ReadWriteOnce` access mode where a PV or PVC can be used on a single node by many pods. RWOP uses SELinux context mount, which allows the driver to mount the volume directly with the correct SELinux labels. This eliminates the need to recursively relabel the volume, and pod startup can be significantly faster. 
```

In {product-title} 4.16, this feature is generally available.

For more information, see xref:..//storage/understanding-persistent-storage.adoc#pv-access-modes_understanding-persistent-storage[Access modes].

[id="ocp-4-16-storage-csi-vsphere-compliance-check"]

> ==== vSphere 3.1 Updated CSI topology requirements

Reviewer 2: ```suggestion
==== vSphere CSI Driver 3.1 Updated CSI topology requirements
```

> To support VMware vSphere Container Storage Interface (CSI) volume provisioning and usage in multi-zonal clusters, the deployment should match certain requirements imposed by CSI driver. These requirements have changed in 3.1.0, and while {product-title} 4.16 accepts both the old and new tagging methods, you should use the new tagging method since VMware considers the old way an invalid configuration, and therefore to prevent problems you should not use it.

Reviewer 2: ```suggestion
To support VMware vSphere Container Storage Interface (CSI) volume provisioning and usage in multi-zonal clusters, the deployment should match certain requirements imposed by CSI driver. These requirements have changed from 3.1.0, and while {product-title} 4.16 accepts both the old and new tagging methods, you should use the new tagging method since VMware considers the old way an invalid configuration, and therefore to prevent problems you should not use it.
```

My only concern is in OCP4.16, we used vSphere CSI Driver 3.1.2. The current wording might inadvertently lead users to believe that we are using version 3.1.0. Please feel free to make any necessary updates to avoid any potential misunderstanding.

Reviewer: I would consider breaking apart this last sentence also.

[id="ocp-4-16-storage"]
=== Storage

[id="ocp-4-16-storage-clone-azure-file"]
==== Volume cloning supported for Azure File (Technology Preview)

> {product-title} 4.16 introduces volume cloning for the Microsoft Azure File Container Storage Interface (CSI) Driver Operator as a Technology Preview feature. Volume cloning duplicates an existing persistent volume (PV) to help protect against data loss in {product-title}.

Reviewer 8: The cloned volume can be re-used for another purpose

For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-azure-file.adoc#persistent-storage-csi-azure-file[Azure File CSI Driver Operator] and xref:..//storage/container_storage_interface/persistent-storage-csi-cloning.adoc#persistent-storage-csi-cloning[CSI volume cloning].

[id="ocp-4-16-storage-node-expansion-secret"]
==== Node Expansion Secret is generally available

> The Node Expansion Secret feature allows your cluster to expand storage volumes, even when access to those volumes requires a secret (for example, a credential for accessing a Storage Area Network (SAN) fabric) to perform the node expand operation. {product-title} 4.16 supports this feature as generally available.

Reviewer 8: I think it's for "mounted" volumes.
Suggest: `The Node Expansion Secret feature allows your cluster to expand storage mounted volumes`
cc @jsafrane

Author: Fixed.

[id="ocp-4-16-vsphere-max-snapshot"]
==== Changing vSphere CSI maximum number of snapshots is generally available
The default maximum number of snapshots in VMWare vSphere Container Storage Interface (CSI) is 3 per volume. In {product-title} 4.16, you can now change this maximum number of snapshots to a maximum of 32 per volume. You also have granular control of the maximum number of snapshots for vSAN and Virtual Volume datastores. {product-title} 4.16 supports this feature as generally available.

> For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-vsphere.adoc#vsphere-change-max-snapshot_persistent-storage-csi-vsphere[Changing the maximum number of snapshots for vSphere]

Reviewer 4: Is the link correct? In preview this leads to vsphere csi driver operator page, but should lead to actual snapshot limit configuration here:

https://github.com/openshift/openshift-docs/pull/74295/files#diff-73b460816ff2db4ab1fb5cb7d16efddc3015860cb41e811acd00131cf7b162c2R8

[id="ocp-4-16-pv-last-phase-transition-time"]
==== Persistent volume last phase transition time parameter (Technology Preview)
In {product-title} 4.16 introduces a new parameter, `LastPhaseTransitionTime`, which has a timestamp that is updated every time a persistent volume (PV) transitions to a different phase (`pv.Status.Phase`). This feature is being released with Technology Preview status.

[id="ocp-4-16-storage-csi-smb-cifs-driver-operator"]

> ==== Persistent storage using SMB CIFS CSI Driver Operator (Technology Preview)

Reviewer 5: ```suggestion
==== Persistent storage using CIFS/SMB CSI Driver Operator (Technology Preview)

```
Maybe better to keep the same name with the display name https://github.com/openshift/csi-operator/blob/79ca0ac2df11f8e047e3eae4e08f13bf745814f1/Dockerfile.samba#L10 .

Reviewer 8: +1

Author: Fixed.

> {Product-title} is capable of provisioning persistent volumes (PVs) using the Container Storage Interface (CSI) driver for the Server Message Block (SMB) protocol Common Internet File System (CIFS) dialect. The SMB CIFS CSI Driver Operator that manages this driver is in Technology Preview status.

Reviewer 8: Sentence sounds a bit strange.
` is capable of provisioning persistent volumes (PVs) with a Container Storage Interface (CSI) driver using the Server Message Block (SMB)...` ?

Author: Fixed.

[id="ocp-4-16-storage"]
=== Storage

[id="ocp-4-16-storage-clone-azure-file"]
==== Volume cloning supported for Azure File (Technology Preview)
{product-title} 4.16 introduces volume cloning for the Microsoft Azure File Container Storage Interface (CSI) Driver Operator as a Technology Preview feature. Volume cloning duplicates an existing persistent volume (PV) to help protect against data loss in {product-title}. You can also use a volume clone just as you would use any standard volume.

For more information, see xref:../storage/container_storage_interface/persistent-storage-csi-azure-file.adoc#persistent-storage-csi-azure-file[Azure File CSI Driver Operator] and xref:..//storage/container_storage_interface/persistent-storage-csi-cloning.adoc#persistent-storage-csi-cloning[CSI volume cloning].

[id="ocp-4-16-storage-node-expansion-secret"]
==== Node Expansion Secret is generally available

> The Node Expansion Secret feature allows your cluster to expand storage mounted to volumes, even when access to those volumes requires a secret (for example, a credential for accessing a Storage Area Network (SAN) fabric) to perform the node expand operation. {product-title} 4.16 supports this feature as generally available.

Author: Done.

|Technology Preview
|Technology Preview

|SMB CIFS CSI Driver Operator
|Not Available
|Not Available
|Technology Preview

|====

> 

Author: @jsafrane @gcharot Should we add this feature to the feature table? I got the impression that we didn't want to too strongly promote this feature.

> 

Author: @radeore We are not wishing to emphasis this feature, so we've decided to not put it in the feature table.

> 

Reviewer 8: @lpettyjo Do we need to keep filestore CSI in the TP table since it's now GA for 3 consecutive releases?

Author: Yes. That's kinda how we do it. For 4.17, it will leave the list.

Reviewer 8: ok all good then !
